### ISSUE TITLE
Markdown parser

### ISSUE THREAD

**Issue by isentropic** [gnab/remark#633]
Dear developers,
Just wanna mention that remark is perhaps better than all the rest of the web presentation frameworks I've seen so far even today. Nothing truly beats remark as it is compact 1 html with plain default theme appropriate for a professional academic environment.

The issue with the parser has been here for way too long though. My issues are:

1. Math does not work properly (underscores and escaping characters)
2. Code listings can behave irregularly

Basically the 2 most commented issues (https://github.com/gnab/remark/issues/416, https://github.com/gnab/remark/issues/336)
These issues persisted for ~4 years, and I think it is the main complaint from users.

It would be great if this was resolved somehow. I can help with docs or some other non-html-js stuff.
